quinn-proto/src/connection/assembler.rs

```diff
@@ -99,32 +107,60 @@ impl Assembler {
         let new = BinaryHeap::with_capacity(self.data.len());
         let old = mem::replace(&mut self.data, new);
         for chunk in old.into_sorted_vec().into_iter().rev() {
-            let end = offset + (buffer.len() as u64);
-            if let Some(overlap) = end.checked_sub(chunk.offset) {
-                if let Some(bytes) = chunk.bytes.get(overlap as usize..) {
-                    buffer.extend_from_slice(bytes);
+            if chunk.size < chunk.allocation_size {
```

Comment:
If you invert this condition we can continue early, which should make this easier to follow (and a smaller diff).

Reply:
OK, done.